### PR TITLE
Make it possible to update and persist changes to Config

### PR DESCRIPTION
**Pull request by spyrkob**
Part of #42 and #41

cc @TomasHofman

### CODE REVIEW

File: prospero-common/src/test/java/org/wildfly/prospero/model/ProvisioningConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addRepositoryThrowsErrorIfSameIdDifferentUrl() throws Exception {
+        provisioningConfig.addRepository(new RepositoryRef("existing", "file:///foo.bar"));
+
+        assertFalse(provisioningConfig.addRepository(new RepositoryRef("existing", "file:///different.url")));
+
+        assertThat(provisioningConfig.getRepositories()).containsExactly(
+                new RepositoryRef("existing", "file:///foo.bar")
+        );
```
Comment (TomasHofman) [inline prospero-common/src/test/java/org/wildfly/prospero/model/ProvisioningConfigTest.java:51]: I think these following assert statements would never be executed, as the exception would be thrown before them.

**Review by TomasHofman — CHANGES_REQUESTED** [review]
Looks good to me, just a small thing in a test case.

File: prospero-common/src/test/java/org/wildfly/prospero/model/ProvisioningConfigTest.java (hunk 1 above)
Comment (spyrkob) [reply prospero-common/src/test/java/org/wildfly/prospero/model/ProvisioningConfigTest.java:51]: Yes that's true, forgot to remove those

File: prospero-common/src/test/java/org/wildfly/prospero/model/ProvisioningConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        assertFalse(provisioningConfig.addRepository(new RepositoryRef("existing", "file:///foo.bar")));
+
+        assertThat(provisioningConfig.getRepositories()).containsExactly(
+                new RepositoryRef("existing", "file:///foo.bar")
+        );
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addRepositoryThrowsErrorIfSameIdDifferentUrl() throws Exception {
+        provisioningConfig.addRepository(new RepositoryRef("existing", "file:///foo.bar"));
```
Comment (TomasHofman) [inline prospero-common/src/test/java/org/wildfly/prospero/model/ProvisioningConfigTest.java:45]: We could check this returned false, to make sure that IllegalArgumentException was not thrown by this line.

Comment (spyrkob) [reply prospero-common/src/test/java/org/wildfly/prospero/model/ProvisioningConfigTest.java:45]: Yeah, I removed `expected` and replaced it with `try/catch` and `Assert.fail`. Clunky, but at least it guarantees to check the right thing

**Comment by spyrkob** [comment]
hmm something funny happened with merging... gonna close this and push last change directly

**Pull request closed without merging**